tests/pthread_tls/tests/01-run.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+
+def _check_test_output(child):
+    child.expect('show tls values:')
+    for i in range(20):
+        if i != 5:
+            child.expect('key\[%d\]: [0-9]*, val: %d'
+                         % (i, i + 1 if i != 3 else 42))
+
+
+def testfunc(child):
+    child.expect('START')
+    child.expect('-= TEST 1 - create 20 tls with sequencial values 0...19 =-')
+    _check_test_output(child)
+    child.expect('-= TEST 2 - '
+                 'delete deliberate key \(key\[5\]:[0-9]{10}\) =-')
+    _check_test_output(child)
+    child.expect('-= TEST 3 - create new tls =-')
+    _check_test_output(child)
+    child.expect('-= TEST 4 - delete all keys =-')
+    child.expect('show tls values:')
+    child.expect('-= TEST 5 - try delete non-existing key =-')
+    child.expect('try to delete returns: 0')
+    child.expect('-= TEST 6 - add key and delete without a tls =-')
+    child.expect('created key: [0-9]{10}')
```

Comment:
Still broken for me because of `[0-9]{10}`, can just be replaced by `\d`.

Reply:
`\d` is just one digit. `\d+` would be any number with at least one digit

Reply:
Ok, I missed a few and my regexp was not super nice. The `\d+` works on native. It's pushed